File: TempServer/ClientObject.cs
using System;
using System.Collections.Generic;
using System.Net.Sockets;
using System.Text;

namespace TempServer
{
    class ClientObject
    {
        const int NAME_LEN = 32;
        const string SERVER = "SERVER";

        public string username = null;

        public TcpClient client;
        NetworkStream stram = null;

        Queue<string> messages = new Queue<string>();

        public delegate void MessageReciver(string sent_from, string send_to, string message);
        public event MessageReciver Recived;

        public ClientObject(TcpClient tcpClient)
        {
            client = tcpClient;
        }

        public void Process()
        {
            Recive();
        }


        public void Recive()
        {
            try
            {
                stram = client.GetStream();
                byte[] data = new byte[64];

                while (true)
                {
                    //recive
                    StringBuilder strBuild = new StringBuilder();
                    int bytes = 0;
                    do
                    {
                        bytes = stram.Read(data, 0, data.Length);
                        strBuild.Append(Encoding.Unicode.GetString(data, 0, bytes));
                    } while (stram.DataAvailable);

                    string message = strBuild.ToString();
                    string send_to = message.Substring(0, NAME_LEN);
                    send_to = send_to.Substring(0, send_to.IndexOf('%'));
                    if (send_to == SERVER)
                    {
                        string command = message.Substring(NAME_LEN, 4);//TODO: 4 isn't beautiful
                        switch (command)
                        {
                            case "AUTH":
                                username = message.Substring(NAME_LEN + 4);
                                string password = username.Substring(username.LastIndexOf('%')+1);
                                username = username.Substring(0,username.IndexOf('%'));

                                if (!Program.AddClient(username, password, Send))
                                    throw new Exception(username + " Denied");
                                break;
                            case "ADD_":
                                string name_to = message.Substring(NAME_LEN+4);
                                Program.AddRequest(username, name_to);
                                break;
                            case "EXIT":
                                throw new Exception(username+" Disconnected");
                                break;
                            case "REG_":
                                string t_username = message.Substring(NAME_LEN + 4,NAME_LEN);
                                string t_password = message.Substring(NAME_LEN + 4 + NAME_LEN);
                                if(Program.RegistrationRequest(t_username, t_password))
                                {
                                    Send("SERVER", t_username, "REG_FINE");
                                    throw new Exception(username + " Registred");
                                }
                                else
                                {
                                    Send("SERVER", t_username, "REG_BAD");
                                    throw new Exception(username + " not registred");
                                }
                                break;
                            default:
                                break;
                        }
                    }
                    else if (username != null)
                    {
                        string sent_from = username;
                        send_to = send_to;
                        message = message.Substring(NAME_LEN);
                        Recived.Invoke(sent_from, send_to, message);
                    }

                }
            }
            catch (Exception ex)
            {
                stram.Close();
                client.Close();
                //Program.RemoveClient(username); //TODO: safety threads;
                Console.WriteLine(ex.Message);

                return;
            }
            finally
            { 
                if(username!=null)
                    Program.RemoveClient(username);

                if (stram != null)
                    stram.Close();
                if (client != null)
                    client.Close();

            }
        }

        public void Send(string sent_from, string send_to, string message)
        {
            if (username != null)
            {
                byte[] data = new byte[64];
                if(sent_from!="SERVER")
                    data = Encoding.Unicode.GetBytes(sent_from+"^:"+message);
                else
                    data = Encoding.Unicode.GetBytes(sent_from + message);
                if (data.Length < 64)
                {
                    byte[] ndata = new byte[64];
                    ndata.Initialize();
                    data.CopyTo(ndata, 0);
                    data = ndata;
                }
                stram.Write(data, 0, data.Length);
            }
        }

    }
}

File: true_server/client_object.py
import socket


class ClientObject:
    __NAME_LEN = 32
    __SERVER = "SERVER"

    __PARENT = None

    username = ""

    on_recived = None

    # username
    # client
    # #TODO: events

    def __init__(self, connection, parent): #parent: Server
        self.client = connection
        self.__PARENT = parent

    def __del__(self):
        self.client.close()

    def process(self):
        self.recive()

    def recive(self):
        try:
            self.client.setblocking(0)
            while True:

                data = b""
                # reciving data
                try:
                    while True:
                        recived = self.client.recv(64)
                        if not recived:
                            break
                        data += recived
                except:
                    if not data:
                        continue

                message = data.decode("unicode_internal")

                send_to = message[:self.__NAME_LEN]
                send_to = send_to[:send_to.find("%")]
                if send_to==self.__SERVER:
                    command = message[self.__NAME_LEN:self.__NAME_LEN+4]
                    if command=="AUTH":
                        self.__PARENT.log("Auth request")
                        self.username = message[self.__NAME_LEN+4:]
                        password = self.username[self.username.rfind("%")+1:]
                        self.username = self.username[:self.username.find("%")]

                        if not self.__PARENT.add_client(self.username, password, self.send):
                            raise Exception(self.username+" Denied")

                    elif command=="ADD_":
                        name_to = message[self.__NAME_LEN+4:]
                        self.__PARENT.add_request(self.username,name_to)

                    elif command=="EXIT":
                        raise Exception(self.username + " Disconnected")

                    elif command=="REG_":
                        t_username = message[self.__NAME_LEN+4:2*self.__NAME_LEN+4]
                        t_password = message[2*self.__NAME_LEN+4:]

                        if self.__PARENT.registration_request(t_username,t_password):
                            self.send(self.__SERVER, t_username, "REG_FINE")
                            raise Exception(self.username +" Registred")

                        else:
                            self.send(self.__SERVER, t_username, "REG_BAD")
                            raise Exception(self.username + "NOT Registred")

                elif self.username:
                    sent_from = self.username
                    send_to = send_to
                    message = message[self.__NAME_LEN:]
                    self.on_recived(sent_from, send_to, message)

        except Exception as ex:
            self.client.close()
            self.__PARENT.log(ex)
            return

        finally:
            if self.username:
                self.__PARENT.remove_client(self.username)

            self.client.close()
            return

    def send(self, sent_from, send_to, message):
        if self.username or sent_from==self.__SERVER:
            if sent_from!=self.__SERVER:
                data = (sent_from+"^"+send_to+":"+message).encode("unicode_internal")
            else:
                data = (sent_from + message).encode("unicode_internal")
            if len(data) < 64:
                while len(data) < 64:
                    data += b'\x00'
            self.client.send(data)

